src/hera/shared/_global_config.py

```diff
@@ -28,17 +30,17 @@ class _GlobalConfig:
 
     # protected attributes are ones that are computed/go through some light processing upon setting or
     # are processed upon accessing. The rest, which use primitive types, such as `str`, can remain public
-    _token: Union[Optional[str], Callable[[], Optional[str]]] = None
+    _token: Union[Optional[str], TokenGenerator, Callable[[], Optional[str]]] = None
```

Comment:
Should the `Optional` be on the outside?
```suggestion
    _token: Optional[Union[str], TokenGenerator, Callable[[], Optional[str]]] = None
```